mrmustard/lab_dev/states/base.py

```diff
@@ -843,11 +843,39 @@ def random(cls, modes, m=None, max_r=1.0):
         max_idx = max(modes)
 
         ancilla = list(range(max_idx + 1, max_idx + m + 1))
-        full_wires = modes + ancilla
+        full_wires = list(modes) + ancilla
 
         psi = Ket.random(full_wires, max_r)
         return psi[modes]
 
+    @property
+    def is_positive(self) -> bool:
+        r"""
+        Whether this DM is a positive operator.
+        """
+        batch_dim = self.representation.A.shape[0]
```

Comment:
```suggestion
        batch_dim = self.representation.ansatz.batch_size
```
If we're making these batch size calls on representation enough we could just expose it at the representation level as well 🤔